File: ios/WebyarNative/Sources/Core/Models/TeamChat.swift
import Foundation

// Operator-to-operator messages — the console calls this the internal inbox,
// and its Colleagues tab. Mirrors `/api/team-chat` and `src/hooks/useTeamChat.ts`.

struct Colleague: Decodable, Identifiable, Hashable, Sendable {
    let userId: String
    let role: String?
    let fullName: String?
    let email: String?
    let avatarURL: String?
    let unread: Int?
    let lastMessage: LastTeamMessage?

    var id: String { userId }

    var displayName: String {
        if let name = fullName, !name.trimmingCharacters(in: .whitespaces).isEmpty { return name }
        if let email, !email.isEmpty { return email }
        return "—"
    }

    struct LastTeamMessage: Decodable, Hashable, Sendable {
        let body: String?
        let createdAt: Date?
        let outgoing: Bool?
        let attachmentKind: String?

        enum CodingKeys: String, CodingKey {
            case body
            case createdAt = "created_at"
            case outgoing
            case attachmentKind = "attachment_kind"
        }
    }

    enum CodingKeys: String, CodingKey {
        case userId = "user_id"
        case role
        case fullName = "full_name"
        case email
        case avatarURL = "avatar_url"
        case unread
        case lastMessage = "last_message"
    }
}

struct ColleaguesResponse: Decodable, Sendable {
    let colleagues: [Colleague]
    let totalUnread: Int?
    let me: String?

    enum CodingKeys: String, CodingKey {
        case colleagues
        case totalUnread = "total_unread"
        case me
    }
}

struct TeamMessage: Decodable, Identifiable, Hashable, Sendable {
    let id: String
    let senderId: String
    let recipientId: String?
    let body: String?
    let attachment: MessageAttachment?
    let readAt: Date?
    let createdAt: Date?

    enum CodingKeys: String, CodingKey {
        case id
        case senderId = "sender_id"
        case recipientId = "recipient_id"
        case body
        case attachment
        case readAt = "read_at"
        case createdAt = "created_at"
    }
}

struct TeamThreadResponse: Decodable, Sendable {
    let messages: [TeamMessage]
    let me: String?
}

File: android/app/src/main/kotlin/com/webyar/operator/core/model/TeamChat.kt
package com.webyar.operator.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.time.Instant

// Operator-to-operator messages — the console calls this the internal inbox,
// and its Colleagues tab. Mirrors `/api/team-chat` and `src/hooks/useTeamChat.ts`.

@Serializable
data class Colleague(
    @SerialName("user_id") val userId: String,
    val role: String? = null,
    @SerialName("full_name") val fullName: String? = null,
    val email: String? = null,
    @SerialName("avatar_url") val avatarUrl: String? = null,
    val unread: Int? = null,
    @SerialName("last_message") val lastMessage: LastTeamMessage? = null,
) {
    val id: String get() = userId

    val displayName: String
        get() = fullName?.takeIf { it.isNotBlank() }
            ?: email?.takeIf { it.isNotEmpty() }
            ?: "—"

    @Serializable
    data class LastTeamMessage(
        val body: String? = null,
        @SerialName("created_at") @Serializable(InstantSerializer::class) val createdAt: Instant? = null,
        val outgoing: Boolean? = null,
        @SerialName("attachment_kind") val attachmentKind: String? = null,
    )
}

@Serializable
data class ColleaguesResponse(
    val colleagues: List<Colleague> = emptyList(),
    @SerialName("total_unread") val totalUnread: Int? = null,
    val me: String? = null,
)

@Serializable
data class TeamMessage(
    val id: String,
    @SerialName("sender_id") val senderId: String,
    @SerialName("recipient_id") val recipientId: String? = null,
    val body: String? = null,
    val attachment: MessageAttachment? = null,
    @SerialName("read_at") @Serializable(InstantSerializer::class) val readAt: Instant? = null,
    @SerialName("created_at") @Serializable(InstantSerializer::class) val createdAt: Instant? = null,
)

@Serializable
data class TeamThreadResponse(
    val messages: List<TeamMessage> = emptyList(),
    val me: String? = null,
)
